### PR TITLE
Rename WFController to ModelController

### PR DESCRIPTION
**Pull request by SYangster**
- Rename for clearer class inheritance:
ModelController-> BaseModelController 
WFController -> ModelController 
- add integration tests

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/advanced/lr-newton-raphson/README.md [hunk 1]
```diff
@@ -74,12 +74,12 @@ class `FedAvgNewtonRaphson`, which can be found
 `FedAvgNewtonRaphson` class inherits from the
```
[the remark this answers is not in the document]
Comment (SYangster) [reply examples/advanced/lr-newton-raphson/README.md:74]: the `FedAvgNewtonRaphson` workflow subclasses `BaseFedAvg` so nothing needs to be changed

**Review by YuanTingHsieh — APPROVED** [review]
LGTM, just make sure to check for all existing places of "model controller" and "wf controller"

**Review by YuanTingHsieh — APPROVED** [review]
Thanks LGTM

**Pull request merged**